### PR TITLE
Helm chart auto update

### PR DESCRIPTION
**Pull request by vlad-diachenko**
**What this PR does / why we need it**:
implemented a pipeline that will run on Loki tag that starts with `v` and will prepare updater config and run updater with prepared config to update Loki helm chart with the released Loki version. 

Example of PR that is generated by this Pipeline : [here](https://github.com/grafana/loki/pull/8328)

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: .drone/drone.jsonnet [hunk 1]
```diff
@@ -672,6 +677,54 @@ local manifest_ecr(apps, archs) = pipeline('manifest-ecr') {
...
+          'if [ "$RELEASE_TAG" != "$latest_version" ]; then echo "Current version $RELEASE_TAG is not the latest version of Loki. The latest version is $latest_version" && exit 78; fi',
+        ],
+      },
+      {
+        name: 'prepare-helm-chart-update-config',
+        image: 'alpine',
+        depends_on: ['check-version-is-latest'],
+        commands: [
+          'RELEASE_TAG=$(./tools/image-tag)',
+          'echo $PLUGIN_CONFIG_TEMPLATE > %s' % configFileName,
```
Comment (jeschkies) [inline .drone/drone.jsonnet:707]: Where's the template defined? We'd also have to increase the Helm chart version.

Comment (jeschkies) [reply .drone/drone.jsonnet:707]: Just saw that it does update the chart version. I'm not sure I understand how this works though.

Comment (vlad-diachenko) [reply .drone/drone.jsonnet:707]: Hey @jeschkies .
It's Grafana's internal tool that does:
1. update values in yaml file
2. bump minor version of helm chart
3. add an entry to changelog.md file
4. re-generate the docs using `helm-docs` tool

File: .drone/drone.jsonnet [hunk 2]
```diff
@@ -3,6 +3,8 @@ local archs = ['amd64', 'arm64', 'arm'];
 
 local build_image_version = std.extVar('__build-image-version');
 
+local drone_updater_plugin_image = 'us.gcr.io/kubernetes-dev/drone/plugins/updater@sha256:cbcb09c74f96a34c528f52bf9b4815a036b11fed65f685be216e0c8b8e84285b';
```
Comment (ssncferreira) [inline .drone/drone.jsonnet:6]: Is this a drone plugin or a specific one for Grafana? 
nit: Should we maybe point to main?

**Review by ssncferreira — APPROVED** [review]
LGTM! Great idea 🎉 
Missing a rebase on main

File: .drone/drone.jsonnet (hunk 2 above)
Comment (vlad-diachenko) [reply .drone/drone.jsonnet:6]: it's drone plugin that we developed https://github.com/grafana/deployment_tools/tree/master/drone/plugins/cmd/updater 

the release process of this plugin provides only `latest` version of the image. so, we decided to use image `sha` to stick to needed version ;)

**Comment by vlad-diachenko** [comment]
added type/bug label to backport changes

**Pull request merged**